opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -233,6 +233,10 @@ namespace Opm {
         // one process.
         wells_active_ = localWellsActive() ? 1 : 0;
         wells_active_ = grid.comm().max(wells_active_);
+        // Number of phases can only be determined globally as it needs
+        // an active well and there might be processes without wells.
+        global_num_phases_ = numPhases();
+        global_num_phases_ = grid.comm().max(global_num_phases_);
```

Comment:
I think `numPhases()` should be same cross all the processes, not depending on whether there is a well there. 

If they are different, then the function is an ill function.